inference-api/inference_api_python_client/client.py

```diff
@@ -0,0 +1,156 @@
+    model_id: str,
+    patient_mrn: str,
+    segmentation_destination_dirname: str,
+  ) -> List[Inference]:
+    """Asynchronous function to fetch inferences from the API matching the model_id and patient_mrn, and
+    download their segmentations locally.
+
+    Args:
+      model_id (str): Model ID for the inferences.
+      patient_mrn (str): Medical record number (MRN) of the patient.
+      segmentation_destination_dirname (str): Directory name where the segmentations will be downloaded.
+
+    Returns:
+      List of Inferences (each in Dict form, loaded from JSON) fetched from the API.
+
+    Raises:
+      InferenceAPIRequestFailedError: If a 400- or 500- response is received from the server.
+      JSONResponseParseError: If a 200- response is received from the server, but it contains invalid JSON.
+      SegmentationDownloadError: If one or more segmentations at the URLs returned by the server fail to download.
+      FailedToWriteSegmentationError: If segmentation_destination_dirname is invalid or lacks write permissions.
+    """
+    resource_path = self.GET_INFERENCE_PATH.format(
+      model_id=model_id,
+      patient_mrn=patient_mrn,
+    )
+    return await self._django_jwt_client.get_json(resource_path, self._session)
+
+  def get_inferences(
+    self,
+    model_id: str,
+    patient_mrn: str,
+    segmentation_destination_dirname: str,
+  ) -> List[Inference]:
+    """Function to fetch inferences from the API matching the model_id and patient_mrn, and
+    download their segmentations locally.
+
+    Args:
+      model_id (str): Model ID for the inferences.
+      patient_mrn (str): Medical record number (MRN) of the patient.
+      segmentation_destination_dirname (str): Directory name where the segmentations will be downloaded.
+
+    Returns:
+      List of Inferences (each in Dict form, loaded from JSON) fetched from the API.
+
+    Raises:
+      InferenceAPIRequestFailedError: If a 400- or 500- response is received from the server.
+      JSONResponseParseError: If a 200- response is received from the server, but it contains invalid JSON.
+      SegmentationDownloadError: If one or more segmentations at the URLs returned by the server fail to download.
+      FailedToWriteSegmentationError: If segmentation_destination_dirname is invalid or lacks write permissions.
+    """
+    return async_to_sync(self.get_inferences_async)(
+      model_id=model_id,
+      patient_mrn=patient_mrn,
+      segmentation_destination_dirname=segmentation_destination_dirname,
+    )
+
+  async def _download_segmentations(
+    self,
+    presigned_urls: str,
+    destination_dirname: str,
+  ) -> None:
+    for presigned_url in presigned_urls:
+      destination_basename = self._get_destination_basename(presigned_url)
+      destination_filename = os.path.join(destination_dirname, destination_basename)
+      async with self._session.get(presigned_url) as response:
+        if response.status >= 400 or not hasattr(response, 'content'):
```

Comment:
I think the recommended way to achieve this check with `aiohttp` is with `if not response.ok`